Compilation Error: Error C2676

Got this when trying to compile the latest commit. The last time I ran a build was commit `2ec2229` and that was successful.
Windows 10
```
# usual command
cmake -B ./build -DGGML_CUDA=ON -DGGML_BLAS=OFF -DGGML_SCHED_MAX_COPIES=1
cmake --build ./build --config Release -j 20
```

Log:
```
  iqk_gemm_kquants.cpp
  iqk_gemm_ktquants.cpp
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_ktquants.cpp(47,61): error C2676: binary '^': '__m256i' does not define t
his operator or a conversion to a type acceptable to the predefined operator [C:\Textgen\ik_llama.cpp\build\ggml\src\gg
ml.vcxproj]
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_ktquants.cpp(83,46): error C2676: binary '^': '__m256i' does not define t
his operator or a conversion to a type acceptable to the predefined operator [C:\Textgen\ik_llama.cpp\build\ggml\src\gg
ml.vcxproj]
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_ktquants.cpp(120,65): error C2676: binary '^': '__m256i' does not define
this operator or a conversion to a type acceptable to the predefined operator [C:\Textgen\ik_llama.cpp\build\ggml\src\g
gml.vcxproj]
  iqk_gemm_iquants.cpp
  iqk_gemm_iqk_quants.cpp
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_iqk_quants.cpp(810,84): warning C4244: 'argument': conversion from 'const
 uint16_t' to 'char', possible loss of data [C:\Textgen\ik_llama.cpp\build\ggml\src\ggml.vcxproj]
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_iqk_quants.cpp(1279,34): message : see reference to function template ins
tantiation '__m256i `anonymous-namespace'::DequantizerIQ2KS::new_block<Q8<1,block_q8_K>>(int,const Q8 &,__m256 *)' bein
g compiled [C:\Textgen\ik_llama.cpp\build\ggml\src\ggml.vcxproj]
          with
          [
              Q8=Q8<1,block_q8_K>
          ]
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_iqk_quants.cpp(2050,1): message : see reference to function template inst
antiation 'void `anonymous-namespace'::mul_mat_qX_K_q8_K_T<Dequantizer,1>(int,const void *,size_t,const DataInfo &,int)
' being compiled [C:\Textgen\ik_llama.cpp\build\ggml\src\ggml.vcxproj]
          with
          [
              Dequantizer=`anonymous-namespace'::DequantizerIQ2KS
          ]
C:\Textgen\ik_llama.cpp\ggml\src\iqk\iqk_gemm_iqk_quants.cpp(2070,13): message : see reference to function template ins
tantiation 'void `anonymous-namespace'::set_functions<`anonymous-namespace'::DequantizerIQ2KS>(std::array<mul_mat_t,8>
&)' being compiled [C:\Textgen\ik_llama.cpp\build\ggml\src\ggml.vcxproj]
  iqk_gemm_1bit.cpp
  iqk_gemm_legacy_quants.cpp
  iqk_quantize.cpp
  Generating Code...

C:\Textgen\ik_llama.cpp>
```

Yep, it compiles and runs fine with that PR. Don't know if this is related but I saw this message come up even though it built:
```
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(554,1): error C3493: 'kBlockSize' cannot be implicit
ly captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\l
lama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(555,1): error C3493: 'kGroupSize' cannot be implicit
ly captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\l
lama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(679,1): error C3493: 'kNg' cannot be implicitly capt
ured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-qu
antize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(694,5): error C2064: term does not evaluate to a fun
ction taking 0 arguments [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(722,1): error C3493: 'kBlockSize' cannot be implicit
ly captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\l
lama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(777,1): error C3493: 'kNumVal' cannot be implicitly
captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llam
a-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(821,5): error C2064: term does not evaluate to a fun
ction taking 0 arguments [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-quantize-stats.vcxproj]
  llama-gguf.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-gguf.exe
  llama-gguf-hash.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-gguf-hash.exe
  llama-bench-matmult.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-bench-matmult.exe
```

No, I am still getting an error, same as last time minus an error for `kBlockSize`:
```
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(555,1): error C3493: 'kGroupSize' cannot be implicit
ly captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\l
lama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(678,1): error C3493: 'kNg' cannot be implicitly capt
ured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-qu
antize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(693,5): error C2064: term does not evaluate to a fun
ction taking 0 arguments [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(780,1): error C3493: 'kNumVal' cannot be implicitly
captured because no default capture mode has been specified [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llam
a-quantize-stats.vcxproj]
C:\Textgen\ik_llama.cpp\examples\quantize-stats\quantize-stats.cpp(824,5): error C2064: term does not evaluate to a fun
ction taking 0 arguments [C:\Textgen\ik_llama.cpp\build\examples\quantize-stats\llama-quantize-stats.vcxproj]
  llama-gguf.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-gguf.exe
  llama-gguf-hash.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-gguf-hash.exe
  llama-bench-matmult.vcxproj -> C:\Textgen\ik_llama.cpp\build\bin\Release\llama-bench-matmult.exe
```

--- a/examples/quantize-stats/quantize-stats.cpp
+++ b/examples/quantize-stats/quantize-stats.cpp
@@ -549,11 +549,15 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
     std::mutex mutex;
     int counter = 0;
     float mse = 0, mse_q = 0;
-    auto compute = [&mutex, &counter, &mse, &mse_q, values, nrows, n_per_row, chunk, block_size = kBlockSize] () {
+    auto compute = [&mutex, &counter, &mse, &mse_q, values, nrows, n_per_row, chunk] () {
+        constexpr int kNumVal = 1 << 15;
+        constexpr int kBlockSize = 32;
+        constexpr int kGroupSize = 8;
+        constexpr int kNg = kBlockSize/kGroupSize;
         double lmse = 0, lmse_q = 0;
-        std::vector<float> scales(n_per_row/block_size);
+        std::vector<float> scales(n_per_row/kBlockSize);
         std::vector<int> best_idx(n_per_row/kGroupSize);
-        std::vector<float> weight(block_size, 1.f);
+        std::vector<float> weight(kBlockSize, 1.f);
         int ncluster = clusters.size() / kGroupSize;
         while (true) {
             std::unique_lock<std::mutex> lock(mutex);
@@ -575,9 +579,10 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 float sigma2 = 0;
                 for (int j = 0; j < n_per_row; ++j) sigma2 += xr[j]*xr[j];
                 sigma2 /= n_per_row;
-                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
-                    auto xb = xr + block_size*ib;
-                    float d = find_best_scale(block_size, xb, weight.data(), iq4k_values, 5);
+                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
+                    auto xb = xr + kBlockSize*ib;
+                    //for (int i = 0; i < kBlockSize; ++i) weight[i] = 0.25f*sigma2 + xb[i]*xb[i];
+                    float d = find_best_scale(kBlockSize, xb, weight.data(), iq4k_values, 5);
                     float id = d ? 1/d : 0.f;
 #ifdef __AVX2__
                     auto vid = _mm256_set1_ps(id);
@@ -662,7 +667,7 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 }
                 float amax_scale = std::abs(scales[0]);
                 float max_scale  = scales[0];
-                for (int ib = 1; ib < n_per_row/block_size; ++ib) {
+                for (int ib = 1; ib < n_per_row/kBlockSize; ++ib) {
                     float ax = std::abs(scales[ib]);
                     if (ax > amax_scale) {
                         amax_scale = ax;
@@ -671,10 +676,10 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 }
                 float d = max_scale/scale_values[0];
                 float id = d ? 1/d : 0.f;
-                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
+                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
                     int ls = best_index_scale(scale_values, id*scales[ib]);
                     float dl = d * scale_values[ls];
-                    auto xb = xr + block_size*ib;
+                    auto xb = xr + kBlockSize*ib;
                     for (int l = 0; l < kNg; ++l) {
                         auto q = codes.data() + kGroupSize*best_idx[ib*kNg+l];
                         for (int k = 0; k < kGroupSize; ++k) {
@@ -688,9 +693,8 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
             }
         }
     };
-    std::vector<std::thread> workers(nthread-1);
+    std::vector<std::thread> workers(nthread);
     for (auto& w : workers) w = std::thread(compute);
-    compute();
     for (auto& w : workers) w.join();
     tot_mse += mse;
     tot_mse_q += mse_q;
@@ -716,15 +720,12 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
     std::mutex mutex;
     int counter = 0;
     float mse = 0, mse_q = 0;
-#ifdef __AVX2__
-    __m256 vx[kBlockSize/8];
-    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk, block_size = kBlockSize, &vx] () {
-#else
-    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk, block_size = kBlockSize] () {
-#endif
+    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk] () {
+        constexpr int kBlockSize = 8;
+        constexpr int kNumVal = 1 << 12;
         float lmse = 0, lmse_q = 0;
-        std::vector<float> scales(n_per_row/block_size);
-        std::vector<int> best_idx(n_per_row/block_size);
+        std::vector<float> scales(n_per_row/kBlockSize);
+        std::vector<int> best_idx(n_per_row/kBlockSize);
         while (true) {
             std::unique_lock<std::mutex> lock(mutex);
             int first = counter; counter += chunk;
@@ -735,6 +736,7 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
             lock.unlock();
             int last = std::min(first + chunk, nrows);
 #ifdef __AVX2__
+            __m256 vx[kBlockSize/8];
             __m256 sqx[8];
             __m256i add_idx = _mm256_set_epi32(7, 6, 5, 4, 3, 2, 1, 0);
             float sx[8];
@@ -742,11 +744,11 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
 #endif
             for (int row = first; row < last; ++row) {
                 auto xr = values + row*n_per_row;
-                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
+                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
                     float best = 0, d = 0; int jbest = -1;
-                    auto xb = xr + block_size*ib;
+                    auto xb = xr + kBlockSize*ib;
 #ifdef __AVX2__
-                    for (int l = 0; l < block_size/8; ++l) {
+                    for (int l = 0; l < kBlockSize/8; ++l) {
                         vx[l] = _mm256_loadu_ps(xb+8*l);
                     }
                     auto vbest = _mm256_set1_ps(0.f);
@@ -755,8 +757,8 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                         auto idx = _mm256_add_epi32(_mm256_set1_epi32(j), add_idx);
                         for (int i = 0; i < 8; ++i) {
                             sqx[i] = _mm256_setzero_ps();
-                            for (int l = 0; l < block_size/8; ++l) {
-                                auto qv = _mm256_loadu_ps(codes.data() + block_size*(j+i) + 8*l);
+                            for (int l = 0; l < kBlockSize/8; ++l) {
+                                auto qv = _mm256_loadu_ps(codes.data() + kBlockSize*(j+i) + 8*l);
                                 sqx[i] = _mm256_fmadd_ps(vx[l], qv, sqx[i]);
                             }
                         }
@@ -772,32 +774,32 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                     for (int j = 1; j < 8; ++j) {
                         if (sx[j] > best) { best = sx[j]; jbest = index[j]; }
                     }
-                    auto qv = codes.data() + block_size*jbest;
+                    auto qv = codes.data() + kBlockSize*jbest;
                     float sumqx = 0;
-                    for (int k = 0; k < block_size; ++k) sumqx += xb[k]*qv[k];
+                    for (int k = 0; k < kBlockSize; ++k) sumqx += xb[k]*qv[k];
                     d = sumqx*sumq2i[jbest];
 #else
                     for (int j = 0; j < kNumVal; ++j) {
                         if (!sumq2i[j]) continue;
-                        auto qv = codes.data() + block_size*j;
+                        auto qv = codes.data() + kBlockSize*j;
                         float sumqx = 0;
-                        for (int k = 0; k < block_size; ++k) sumqx += qv[k]*xb[k];
+                        for (int k = 0; k < kBlockSize; ++k) sumqx += qv[k]*xb[k];
                         if (sumqx*sumqx*sumq2i[j] > best) {
                             d = sumqx*sumq2i[j]; best = d*sumqx; jbest = j;
                         }
                     }
-                    auto qv = codes.data() + block_size*jbest;
+                    auto qv = codes.data() + kBlockSize*jbest;
 #endif
                     scales[ib] = d;
                     best_idx[ib] = jbest;
-                    for (int k = 0; k < block_size; ++k) {
+                    for (int k = 0; k < kBlockSize; ++k) {
                         float diff = xb[k] - d*qv[k];
                         lmse += diff*diff;
                     }
                 }
                 float amax_scale = std::abs(scales[0]);
                 float max_scale  = scales[0];
-                for (int ib = 1; ib < n_per_row/block_size; ++ib) {
+                for (int ib = 1; ib < n_per_row/kBlockSize; ++ib) {
                     float ax = std::abs(scales[ib]);
                     if (ax > amax_scale) {
                         amax_scale = ax;
@@ -806,12 +808,12 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                 }
                 float d = max_scale/scale_values[0];
                 float id = d ? 1/d : 0.f;
-                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
+                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
                     int ls = best_index_scale(scale_values, id*scales[ib]);
                     float dl = d * scale_values[ls];
-                    auto xb = xr + block_size*ib;
-                    auto qv = codes.data() + block_size*best_idx[ib];
-                    for (int k = 0; k < block_size; ++k) {
+                    auto xb = xr + kBlockSize*ib;
+                    auto qv = codes.data() + kBlockSize*best_idx[ib];
+                    for (int k = 0; k < kBlockSize; ++k) {
                         float diff = xb[k] - dl*qv[k];
                         lmse_q += diff*diff;
                     }
@@ -819,9 +821,8 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
             }
         }
     };
-    std::vector<std::thread> workers(nthread-1);
+    std::vector<std::thread> workers(nthread);
     for (auto& w : workers) w = std::thread(compute);
-    compute();
     for (auto& w : workers) w.join();
     tot_mse += mse;
     tot_mse_q += mse_q;

--- a/examples/quantize-stats/quantize-stats.cpp
+++ b/examples/quantize-stats/quantize-stats.cpp
@@ -549,11 +549,11 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
     std::mutex mutex;
     int counter = 0;
     float mse = 0, mse_q = 0;
-    auto compute = [&mutex, &counter, &mse, &mse_q, values, nrows, n_per_row, chunk] () {
+    auto compute = [&mutex, &counter, &mse, &mse_q, values, nrows, n_per_row, chunk, block_size = kBlockSize] () {
         double lmse = 0, lmse_q = 0;
-        std::vector<float> scales(n_per_row/kBlockSize);
+        std::vector<float> scales(n_per_row/block_size);
         std::vector<int> best_idx(n_per_row/kGroupSize);
-        std::vector<float> weight(kBlockSize, 1.f);
+        std::vector<float> weight(block_size, 1.f);
         int ncluster = clusters.size() / kGroupSize;
         while (true) {
             std::unique_lock<std::mutex> lock(mutex);
@@ -575,10 +575,9 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 float sigma2 = 0;
                 for (int j = 0; j < n_per_row; ++j) sigma2 += xr[j]*xr[j];
                 sigma2 /= n_per_row;
-                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
-                    auto xb = xr + kBlockSize*ib;
-                    //for (int i = 0; i < kBlockSize; ++i) weight[i] = 0.25f*sigma2 + xb[i]*xb[i];
-                    float d = find_best_scale(kBlockSize, xb, weight.data(), iq4k_values, 5);
+                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
+                    auto xb = xr + block_size*ib;
+                    float d = find_best_scale(block_size, xb, weight.data(), iq4k_values, 5);
                     float id = d ? 1/d : 0.f;
 #ifdef __AVX2__
                     auto vid = _mm256_set1_ps(id);
@@ -663,7 +662,7 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 }
                 float amax_scale = std::abs(scales[0]);
                 float max_scale  = scales[0];
-                for (int ib = 1; ib < n_per_row/kBlockSize; ++ib) {
+                for (int ib = 1; ib < n_per_row/block_size; ++ib) {
                     float ax = std::abs(scales[ib]);
                     if (ax > amax_scale) {
                         amax_scale = ax;
@@ -672,10 +671,10 @@ static void analyze_x_v2(const char * name, int nrows, int n_per_row, const floa
                 }
                 float d = max_scale/scale_values[0];
                 float id = d ? 1/d : 0.f;
-                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
+                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
                     int ls = best_index_scale(scale_values, id*scales[ib]);
                     float dl = d * scale_values[ls];
-                    auto xb = xr + kBlockSize*ib;
+                    auto xb = xr + block_size*ib;
                     for (int l = 0; l < kNg; ++l) {
                         auto q = codes.data() + kGroupSize*best_idx[ib*kNg+l];
                         for (int k = 0; k < kGroupSize; ++k) {
@@ -717,10 +716,15 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
     std::mutex mutex;
     int counter = 0;
     float mse = 0, mse_q = 0;
-    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk] () {
+#ifdef __AVX2__
+    __m256 vx[kBlockSize/8];
+    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk, block_size = kBlockSize, &vx] () {
+#else
+    auto compute = [&mutex, &counter, &mse, &mse_q, &codes, &sumq2i, values, nrows, n_per_row, chunk, block_size = kBlockSize] () {
+#endif
         float lmse = 0, lmse_q = 0;
-        std::vector<float> scales(n_per_row/kBlockSize);
-        std::vector<int> best_idx(n_per_row/kBlockSize);
+        std::vector<float> scales(n_per_row/block_size);
+        std::vector<int> best_idx(n_per_row/block_size);
         while (true) {
             std::unique_lock<std::mutex> lock(mutex);
             int first = counter; counter += chunk;
@@ -731,7 +735,6 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
             lock.unlock();
             int last = std::min(first + chunk, nrows);
 #ifdef __AVX2__
-            __m256 vx[kBlockSize/8];
             __m256 sqx[8];
             __m256i add_idx = _mm256_set_epi32(7, 6, 5, 4, 3, 2, 1, 0);
             float sx[8];
@@ -739,11 +742,11 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
 #endif
             for (int row = first; row < last; ++row) {
                 auto xr = values + row*n_per_row;
-                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
+                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
                     float best = 0, d = 0; int jbest = -1;
-                    auto xb = xr + kBlockSize*ib;
+                    auto xb = xr + block_size*ib;
 #ifdef __AVX2__
-                    for (int l = 0; l < kBlockSize/8; ++l) {
+                    for (int l = 0; l < block_size/8; ++l) {
                         vx[l] = _mm256_loadu_ps(xb+8*l);
                     }
                     auto vbest = _mm256_set1_ps(0.f);
@@ -752,8 +755,8 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                         auto idx = _mm256_add_epi32(_mm256_set1_epi32(j), add_idx);
                         for (int i = 0; i < 8; ++i) {
                             sqx[i] = _mm256_setzero_ps();
-                            for (int l = 0; l < kBlockSize/8; ++l) {
-                                auto qv = _mm256_loadu_ps(codes.data() + kBlockSize*(j+i) + 8*l);
+                            for (int l = 0; l < block_size/8; ++l) {
+                                auto qv = _mm256_loadu_ps(codes.data() + block_size*(j+i) + 8*l);
                                 sqx[i] = _mm256_fmadd_ps(vx[l], qv, sqx[i]);
                             }
                         }
@@ -769,32 +772,32 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                     for (int j = 1; j < 8; ++j) {
                         if (sx[j] > best) { best = sx[j]; jbest = index[j]; }
                     }
-                    auto qv = codes.data() + kBlockSize*jbest;
+                    auto qv = codes.data() + block_size*jbest;
                     float sumqx = 0;
-                    for (int k = 0; k < kBlockSize; ++k) sumqx += xb[k]*qv[k];
+                    for (int k = 0; k < block_size; ++k) sumqx += xb[k]*qv[k];
                     d = sumqx*sumq2i[jbest];
 #else
                     for (int j = 0; j < kNumVal; ++j) {
                         if (!sumq2i[j]) continue;
-                        auto qv = codes.data() + kBlockSize*j;
+                        auto qv = codes.data() + block_size*j;
                         float sumqx = 0;
-                        for (int k = 0; k < kBlockSize; ++k) sumqx += qv[k]*xb[k];
+                        for (int k = 0; k < block_size; ++k) sumqx += qv[k]*xb[k];
                         if (sumqx*sumqx*sumq2i[j] > best) {
                             d = sumqx*sumq2i[j]; best = d*sumqx; jbest = j;
                         }
                     }
-                    auto qv = codes.data() + kBlockSize*jbest;
+                    auto qv = codes.data() + block_size*jbest;
 #endif
                     scales[ib] = d;
                     best_idx[ib] = jbest;
-                    for (int k = 0; k < kBlockSize; ++k) {
+                    for (int k = 0; k < block_size; ++k) {
                         float diff = xb[k] - d*qv[k];
                         lmse += diff*diff;
                     }
                 }
                 float amax_scale = std::abs(scales[0]);
                 float max_scale  = scales[0];
-                for (int ib = 1; ib < n_per_row/kBlockSize; ++ib) {
+                for (int ib = 1; ib < n_per_row/block_size; ++ib) {
                     float ax = std::abs(scales[ib]);
                     if (ax > amax_scale) {
                         amax_scale = ax;
@@ -803,12 +806,12 @@ static void analyze_x(const char * name, int nrows, int n_per_row, const float *
                 }
                 float d = max_scale/scale_values[0];
                 float id = d ? 1/d : 0.f;
-                for (int ib = 0; ib < n_per_row/kBlockSize; ++ib) {
+                for (int ib = 0; ib < n_per_row/block_size; ++ib) {
                     int ls = best_index_scale(scale_values, id*scales[ib]);
                     float dl = d * scale_values[ls];
-                    auto xb = xr + kBlockSize*ib;
-                    auto qv = codes.data() + kBlockSize*best_idx[ib];
-                    for (int k = 0; k < kBlockSize; ++k) {
+                    auto xb = xr + block_size*ib;
+                    auto qv = codes.data() + block_size*best_idx[ib];
+                    for (int k = 0; k < block_size; ++k) {
                         float diff = xb[k] - dl*qv[k];
                         lmse_q += diff*diff;
                     }

--- a/ggml/src/iqk/iqk_gemm_ktquants.cpp
+++ b/ggml/src/iqk/iqk_gemm_ktquants.cpp
@@ -21,32 +21,6 @@ static inline uint32_t trellis_next(uint32_t& val) {
     return (val & kmask) ^ km32;
 }
 
-static inline __m256i trellis_next8(uint32_t val) {
-    constexpr uint32_t kmask = 0x8fff8fff;
-    constexpr uint32_t km32 = 0x3b603b60;
-    constexpr uint32_t ka = 89226354;
-    constexpr uint32_t kb = 64248484;
-    constexpr uint32_t ka1 = ka*ka;
-    constexpr uint32_t kb1 = kb*ka+kb;
-    constexpr uint32_t ka2 = ka1*ka;
-    constexpr uint32_t kb2 = kb1*ka+kb;
-    constexpr uint32_t ka3 = ka2*ka;
-    constexpr uint32_t kb3 = kb2*ka+kb;
-    constexpr uint32_t ka4 = ka3*ka;
-    constexpr uint32_t kb4 = kb3*ka+kb;
-    constexpr uint32_t ka5 = ka4*ka;
-    constexpr uint32_t kb5 = kb4*ka+kb;
-    constexpr uint32_t ka6 = ka5*ka;
-    constexpr uint32_t kb6 = kb5*ka+kb;
-    constexpr uint32_t ka7 = ka6*ka;
-    constexpr uint32_t kb7 = kb6*ka+kb;
-    __m256i mka = _mm256_setr_epi32(ka, ka1, ka2, ka3, ka4, ka5, ka6, ka7);
-    __m256i mkb = _mm256_setr_epi32(kb, kb1, kb2, kb3, kb4, kb5, kb6, kb7);
-    __m256i mval = _mm256_set1_epi32(val);
-    __m256i mres = _mm256_add_epi32(_mm256_mullo_epi32(mval, mka), mkb);
-    return _mm256_and_si256(mres, _mm256_set1_epi32(kmask)) ^ _mm256_set1_epi32(km32);
-}
-
 static inline float trellis_gen(uint32_t& val, uint32_t* s) {
     const ggml_fp16_t * h = (const ggml_fp16_t *)s;
     s[0] = trellis_next(val);
@@ -80,7 +54,7 @@ struct Trellis1 {
     inline __m256i next8(uint32_t val) const {
         auto mval = _mm256_set1_epi32(val);
         auto mres = _mm256_add_epi32(_mm256_mullo_epi32(mval, mka), mkb);
-        return _mm256_and_si256(mres, mask1) ^ mask2;
+        return _mm256_xor_si256(_mm256_and_si256(mres, mask1), mask2);
     }
 };
 
@@ -117,7 +91,7 @@ struct Trellis2 {
     inline __m256i next8(uint32_t val1, uint32_t val2) {
         __m256i mval = _mm256_setr_epi32(val1, val1, val1, val1, val2, val2, val2, val2);
         __m256i mres = _mm256_add_epi32(_mm256_mullo_epi32(mval, mka), mkb);
-        return _mm256_and_si256(mres, _mm256_set1_epi32(kmask)) ^ _mm256_set1_epi32(km32);
+        return _mm256_xor_si256(_mm256_and_si256(mres, _mm256_set1_epi32(kmask)), _mm256_set1_epi32(km32));
     }
 };
 

